diff --git app-catalog/src/components/releases/EditorDialog.tsx app-catalog/src/components/releases/EditorDialog.tsx
index efda7e2..09091b8 100644
--- app-catalog/src/components/releases/EditorDialog.tsx
+++ app-catalog/src/components/releases/EditorDialog.tsx
@@ -16,6 +16,7 @@ import { useEffect, useRef, useState } from 'react';
 import semver from 'semver';
 import { fetchChart, getActionStatus, upgradeRelease } from '../../api/releases';
 import { jsonToYAML, yamlToJSON } from '../../helpers';
+import {APP_CATALOG_HELM_REPOSITORY} from "../charts/List";
 
 export function EditorDialog(props: {
   openEditor: boolean;
@@ -64,7 +65,8 @@ export function EditorDialog(props: {
         let response;
         let error: Error | null = null;
         try {
-          response = await fetchChart(release.chart.metadata.name)
+          const metadataName = release.chart.metadata.name === APP_CATALOG_HELM_REPOSITORY ? '/' + release.chart.metadata.name : release.chart.metadata.name;
+          response = await fetchChart(metadataName)
         } catch (err) {
           error = err;
         }
